app/assets/javascripts/fae/form/_validator.js

```diff
@@ -69,7 +72,7 @@ Fae.form.validator = {
    * Tests a forms validation after all validation checks have responded
    * Polls validations responses every 50ms to allow uniqueness AJAX calls to complete
    */
-  testValidation: function($this) {
+  testValidation: function($this, $scope) {
```

Comment:
This `$scope` argument seems misnamed, unless it's used for another feature in the pipeline that I'm unaware of. For better maintainability, it should read as a boolean like it's used, i.e. `shouldReturnToTopOnSuccess`. In the function you're adding this to (`formValidate`), it will always be `true`; the value of the variable doesn't need to be passed on.

Reply:
It's not pretty or elegant. but I needed a convenient way to handle cases where $scope is the normal document body and cases where it is an AJAX loaded nested form. The latter was previously being completely ignored by these functions and thus providing no front end validation of an AJAX loaded nested form. If there are form errors in your nested form, then scrolling all the way to the top of the parent form is not an ideal user experience. I can take a stab at renaming and refactoring this though.